cmake/kokkos_arch.cmake

```
@@ -484,6 +484,43 @@ IF(KOKKOS_ENABLE_CUDA AND NOT CUDA_ARCH_ALREADY_SPECIFIED)
     ${CMAKE_CURRENT_SOURCE_DIR}/cmake/compile_tests/cuda_compute_capability.cc
     COMPILE_DEFINITIONS -DSM_ONLY
     RUN_OUTPUT_VARIABLE _CUDA_COMPUTE_CAPABILITY)
+
+  # if user is using kokkos_compiler_launcher, above will fail.
+  IF(NOT _COMPILE_RESULT OR NOT _RESULT EQUAL 0 OR FLAG_INDEX EQUAL -1)
```

```suggestion
  IF(NOT _COMPILE_RESULT OR NOT _RESULT EQUAL 0)
```
Why do we need to check for `FLAG_INDEX` here? We only set a value below.

---

I just inverted the success check

---

Yes, I know but it doesn't make sense to check `FLAG_INDEX` when it's not even defined here yet.